server/testserver.go

```diff
@@ -63,6 +63,22 @@ func StartTestServer(t util.Tester) *TestServer {
 	return s
 }
 
+// StartTestServerJoining starts an in-memory test server that attempts to join `other`.
+func StartTestServerJoining(t util.Tester, other *TestServer) *TestServer {
+	s := &TestServer{Ctx: NewTestContext()}
+	a := other.ServingAddr()
```

Comment:
oh, yeah, I split it out when it was panicking to get a more useful line number. I can put it back inline.